### PR TITLE
Refactor dispatch structure so fallback code lives inline.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26719 Tests for fallback boxed dispatch.
* #26368 Add boxed fallback
* **#26367 Refactor dispatch structure so fallback code lives inline.**
* #26857 Change calling convention of ATenDispatch from getOp to callUnboxed.

This is necessary for boxed fallback, as boxed fallback must
live inside the templated code.  Error reporting code never
has to be in templated code, so that stays in the C++ file.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D17448556](https://our.internmc.facebook.com/intern/diff/D17448556)

### CODE REVIEW

File: aten/src/ATen/core/ATenDispatch.cpp [hunk 1]
```diff
@@ -7,37 +7,31 @@ ATenDispatch & globalATenDispatch() {
...
-  if (function_table_[static_cast<int64_t>(TensorTypeId::UndefinedTensorId)] == nullptr) {
-    std::ostringstream oss;
-    bool first = true;
-    for (int64_t i = 0; i < static_cast<int64_t>(TensorTypeId::NumTensorIds); i++) {
-      if (function_table_[i] != nullptr) {
-        if (!first) oss << ", ";
-        oss << toString(static_cast<TensorTypeId>(i));
-        first = false;
-      }
+void ATenOpTable::reportError(TensorTypeId tid) const {
```
Comment (smessmer) [inline aten/src/ATen/core/ATenDispatch.cpp]: this doesn't seem to report anything? It just discards the constructed string?

**Review by smessmer — APPROVED** [review]

File: aten/src/ATen/core/ATenDispatch.cpp (hunk 1 above)
Comment (ezyang) [reply aten/src/ATen/core/ATenDispatch.cpp]: It raises an exception.

Comment (smessmer) [reply aten/src/ATen/core/ATenDispatch.cpp]: oh I somehow thought line 19 was ending this function already, missed that it continues

**Pull request closed without merging**